firefox_media_tests/playback/test_video_playback.py

```diff
@@ -41,38 +44,25 @@ def test_playback_starts(self):
                                         level='WARNING')
 
     def test_video_playback_partial(self):
-        self.run_playback(timeout=120)
+        self.run_playback(duration=60)
 
     def test_video_playback_full(self):
         self.run_playback()
 
-    def run_playback(self, timeout=0):
+    def run_playback(self, interval=1, duration=None,
+                     stall_wait_time=10):
         with self.marionette.using_context('content'):
             for url in self.test_urls:
-                video = VideoPuppeteer(self.marionette, url)
-                # Some vendors don't start the videos at the beginning,
-                # remembering where you were last time you watched.
-                start_time = video.current_time
+                self.logger.info(url)
+                video = VideoPuppeteer(self.marionette, url,
+                                       interval=interval, duration=duration,
+                                       stall_wait_time=stall_wait_time)
                 verbose_until(Wait(video, timeout=30), video,
                               playback_started)
-                duration_timeout = video.duration * 1.3
-                time_delta = 2
-                if timeout <= 0:
-                    duration = duration_timeout
-                else:
-                    # Looser constraint for partial playback
-                    time_delta = 5
-                    duration = min(timeout, duration_timeout)
 
                 try:
-                    verbose_until(Wait(video, interval=1, timeout=duration),
+                    verbose_until(Wait(video, interval=interval,
+                                       timeout=video.duration+stall_wait_time),
```

Comment:
We can wait and see how the test behaves in production. 

Increasing the timeout does no harm and might help with debugging by distinguishing the `TimeoutException` raised by `verbose_until` from the `VideoException` raised by `playback_done`. (Thinking out loud again: If `stall_wait_time` is exceeded, a `VideoException` is raised. A `TimeoutException` should never occur, because the `VideoException` is always hit first, so, given a longer timeout, if we see a `TimeoutException`, something is wrong with the stall-detection code or there's an issue we have not anticipated.)